app/models/notification.rb

```diff
@@ -0,0 +1,55 @@
+class Notification < ActiveRecord::Base
```

Comment:
* email is optional, I just divide notify into 'with a given activity record' and 'without a given activity record'. I will check if types have an email key words in these two situation.
* Where am I supposed to put those in?   **view/activities**, **view/popups** and **view/notification_mailer**?
* Existing notifications? Means `popup_level` 'popup_achievement' and 'activities shown in course page'?